### PR TITLE
Add Blip Diffusion tutorial

### PR DESCRIPTION
**Pull request by itrushkin**
- [x] Subject-driven generation
- [x] Structure-controlled generation (stylization)
- [x] Subject-driven editing

CVS-121991

### CODE REVIEW

File: notebooks/257-blip-diffusion-subject-generation/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Subject-driven image generation and editing using BLIP Diffusion and OpenVINO
+![](https://github.com/salesforce/LAVIS/raw/main/projects/blip-diffusion/teaser-website.png)
```
Comment (igor-davidyuk) [inline notebooks/257-blip-diffusion-subject-generation/README.md:2]: please test in colab and add badges if it works

**Comment by itrushkin** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1354/discussion/#comment-1752763310'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by itrushkin** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1354/discussion/#comment-1752763336'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by itrushkin** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1354/discussion/#comment-1752763364'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: notebooks/257-blip-diffusion-subject-generation/README.md (hunk 1 above)
Comment (itrushkin) [reply notebooks/257-blip-diffusion-subject-generation/README.md:2]: It does not work in Colab because of RAM limit.

**Comment by itrushkin** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1354/discussion/#comment-1761136123'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: notebooks/257-blip-diffusion-subject-generation/257-blip-diffusion-subject-generation.ipynb
Comment (aleksandr-mokrov) [inline notebooks/257-blip-diffusion-subject-generation/257-blip-diffusion-subject-generation.ipynb:1]: `Convert model to OpenVINO Intermediate Representation (IR)` -> `Convert the model to OpenVINO Intermediate Representation (IR)`

[the remark this answers is not in the document]
Comment (itrushkin) [reply notebooks/257-blip-diffusion-subject-generation/257-blip-diffusion-subject-generation.ipynb:1]: Done

**Comment by eaidova** [comment]
<p>Regerding TODO, diffusers 0.21.4 was release 2 weeks ago, while your referenced PR was merged in july, so maybe it is time to update package?</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1354/discussion/#comment-1761136123'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by itrushkin** [comment]
<p>The referenced PR was merged on September, 21. The latest (v0.21.4) release tag does not contain the required pipeline file: <a href="https://github.com/huggingface/diffusers/blob/v0.21.4/src/diffusers/pipelines/blip_diffusion/pipeline_blip_diffusion.py" target="_blank">https://github.com/huggingface/diffusers/blob/v0.21.4/src/diffusers/pipelines/blip_diffusion/pipeline_blip_diffusion.py</a></p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1354/discussion/#comment-1761136123'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by itrushkin** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1354/discussion/#comment-1764282685'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by itrushkin** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1354/discussion/#comment-1764282656'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: notebooks/258-blip-diffusion-subject-generation/258-blip-diffusion-subject-generation.ipynb
Comment (aleksandr-mokrov) [inline notebooks/258-blip-diffusion-subject-generation/258-blip-diffusion-subject-generation.ipynb:1]: Please add a description how to convert a model to IR format, save and compile it.

**Comment by eaidova** [comment]
@itrushkin I suppose you disabale notebook on ci due to some memory issues. Please also add it into https://github.com/openvinotoolkit/openvino_notebooks/blob/main/.ci/ignore_convert_execution.txt
for disabling also running during conversion notebook in docs

File: notebooks/258-blip-diffusion-subject-generation/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+## Notebook contents
+The tutorial consists of the following steps:
+
+- Prerequisites
+- Load the model
+- Infer original model
+- Convert model to OpenVINO Intermediate Representation (IR)
+- Inference
+- Interactive inference
+
```
Comment (eaidova) [inline notebooks/258-blip-diffusion-subject-generation/README.md:28]: can you please add some demonstration of notebook work and some description which use cases covered in notebook in this section?

File: notebooks/258-blip-diffusion-subject-generation/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+Second, Blip Diffusion demonstrates a remarkable ability to produce diverse images. Unlike models reliant on fixed image templates, Blip Diffusion crafts images from scratch by leveraging the multimodal representation derived from the text prompt, fostering uniqueness in its outputs.
+
+Third, Blip Diffusion excels at generating realistic images due to its training on a dataset containing real-world images.
+
+In summary, Blip Diffusion stands as a potent text-to-image diffusion model for generating high-quality images from text prompts. It distinguishes itself through its capacity to produce images that align closely with the text prompt's subject, its versatility in generating diverse visuals from scratch, and its ability to craft realistic images. In this tutorial we will show how to convert Blip Diffusion model to OpenVINO Intermediate Representation and perform inference.
```
Comment (eaidova) [inline notebooks/258-blip-diffusion-subject-generation/README.md:17]: I believe starting to "In this tutorial ... " is separated thoughts and will be better to split it in separated paragraph for making it more visible

File: notebooks/258-blip-diffusion-subject-generation/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+Once the diffusion model completes its training, it becomes capable of generating images from text prompts. The image generation process commences with a random noise image, progressively integrating more details until it faithfully mirrors the multimodal representation inherent in the text prompt.
+
+Blip Diffusion offers several advantages over other text-to-image diffusion models. First, it excels at generating images that harmonize with the subject matter of the text prompt, owing to its pre-training on a multimodal dataset of text and images, which facilitates the understanding of the relationship between textual and visual elements.
+
+Second, Blip Diffusion demonstrates a remarkable ability to produce diverse images. Unlike models reliant on fixed image templates, Blip Diffusion crafts images from scratch by leveraging the multimodal representation derived from the text prompt, fostering uniqueness in its outputs.
+
+Third, Blip Diffusion excels at generating realistic images due to its training on a dataset containing real-world images.
```
Comment (eaidova) [inline notebooks/258-blip-diffusion-subject-generation/README.md:15]: I think it is very long description with a lot of details about training that makes for users who does not know what this model do reading of readme complicated and may lead to loosing context (our focus is inference, not training, right?). Can we simplify description in readme and stay these details for notebook (maybe with subtitle about model)?

File: notebooks/258-blip-diffusion-subject-generation/258-blip-diffusion-subject-generation.ipynb
Comment (itrushkin) [reply notebooks/258-blip-diffusion-subject-generation/258-blip-diffusion-subject-generation.ipynb:1]: Done

[the remark this answers is not in the document]
Comment (itrushkin) [reply notebooks/258-blip-diffusion-subject-generation/258-blip-diffusion-subject-generation.ipynb:1]: Model had fixed input for text. Now both parameters can accept strings of any length

File: notebooks/258-blip-diffusion-subject-generation/README.md (hunk 2 above)
Comment (itrushkin) [reply notebooks/258-blip-diffusion-subject-generation/README.md:28]: Added images with inference results

File: notebooks/258-blip-diffusion-subject-generation/README.md (hunk 3 above)
Comment (itrushkin) [reply notebooks/258-blip-diffusion-subject-generation/README.md:17]: Removed this section

File: notebooks/258-blip-diffusion-subject-generation/README.md (hunk 4 above)
Comment (itrushkin) [reply notebooks/258-blip-diffusion-subject-generation/README.md:15]: Simplified

**Comment by eaidova** [comment]
@itrushkin 
now github can not open your readme file:

image

I believe it is due to images in base64 format, could you please insert link instead?

**Review by eaidova — APPROVED** [review]

**Review by igor-davidyuk — APPROVED** [review]
looks good

**Pull request merged**